Fabric.Terminology.API/Scripts/Terminology-Intall-Utilities.psm1

```diff
@@ -0,0 +1,135 @@
+Class TerminologyConfig
+{
+    [String]$siteName
+    [String]$appName
+    [String]$sqlServerAddress
+    [String]$appPool
+    [String]$applicationEndpoint
+    [String]$discoveryServiceUrl
+    [PSCredential]$iisUserCredentials
+}
+function Import-Modules() {
+    # Dos Install Utilities
+    $dosInstallUtilities = Get-Childitem -Path ./**/DosInstallUtilities.psm1 -Recurse
+    if($dosInstallUtilities.length -eq 0) {
+        Write-Warning "could not find dos install utilities. Manually installing "
+        Install-Module DosInstallUtilities -Scope CurrentUser
+        Import-Module DosInstallUtilities -Force
+    } else {
+        Import-Module -Name $dosInstallUtilities.FullName -Force
+    }
+	
+	# Fabric Install Utilities
+	$fabricInstallUtilities = ".\Fabric-Install-Utilities.psm1"
+    if(!(Test-Path $fabricInstallUtilities -PathType Leaf)){
+        Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/InstallScripts/master/common/Fabric-Install-Utilities.psm1 -Headers @{"Cache-Control"="no-cache"} -OutFile $dosInstallUtilities
+    }
+    Import-Module -Name $fabricInstallUtilities -Force
+}
+
+function Get-TerminologyConfig{
+    param(
+        [string] $Path
+    )
+    
+    $installSettings = Get-InstallationSettings "terminology"
+
+    $iisUser = Read-Host "Press Enter to accept the default IIS App Pool User"
```

Comment:
Perhaps a better pattern here would be to see if $installSettings.UserName (or whatever the setting is called) is null and only prompting if it's null?  That way Platform Ops can fill in the config file if they want to avoid prompts.

Reply:
I have plans to do that, but as it stands the dos installer "fabric registration" step is broken until I get the Terminology service installed and registered with the discovery service. That is one of my next steps

Reply:
@benjanderson FYI -  As long as the appsettings.json is updated with the DiscoveryServiceUrl the API should register itself with the DiscoveryService when it bootstraps.  It should happen when you do your first ping.  Is it required by the installer for any other reason?